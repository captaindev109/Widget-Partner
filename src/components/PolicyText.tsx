import React, { FC } from 'react';
import Text from '@/generics/Text';
import LinkText from '@/generics/LinkText';

export interface PolicyTextProps {
  text?: string;
  linkText?: string;
  linkUrl?: string;
}

const PolicyText: FC<PolicyTextProps> = ({ text, linkText, linkUrl }) => {
  return (
    <Text variant="Caption" color="gray500" data-testid="policy-text">
      {text}
      <LinkText data-testid="policy-link-text" href={linkUrl} target="_blank" rel="noopener noreferrer">
        {` ${linkText}`}
      </LinkText>
      .
    </Text>
  );
};

PolicyText.defaultProps = {
  text: 'Genom att klicka “Få ett prisförslag” accepterar du att du har tagit del av information om.',
  linkText: 'Hedvigs personuppgiftpolicy',
  linkUrl: 'https://www.hedvig.com/se/personuppgifter',
};

export default PolicyText;
