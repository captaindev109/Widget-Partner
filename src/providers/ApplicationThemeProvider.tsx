import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
