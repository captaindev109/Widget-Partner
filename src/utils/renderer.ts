import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
