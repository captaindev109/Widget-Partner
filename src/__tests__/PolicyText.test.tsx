import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import defaultTheme from '@/theme/theme';
import PolicyText from '@/components/PolicyText';

const text = 'text';
const linkText = 'linkText';
const url = 'https://www.hedvig.com/se/personuppgifter';

describe('Test cases of the PolicyText component', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <PolicyText text={text} linkText={linkText} linkUrl={url} />
      </ApplicationThemeProvider>
    );
  });

  test('to see if the component render without crashing', () => {
    render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <PolicyText text={text} linkText={linkText} linkUrl={url} />
      </ApplicationThemeProvider>
    );
  });

  test('to see if the texts are correct', () => {
    const { getByTestId } = renderedComponent;
    const linkElement = getByTestId('policy-link-text');
    const textElement = getByTestId('policy-text');

    expect(linkElement.textContent).toContain(linkText);
    expect(textElement.textContent).toContain(text);
    expect(linkElement).toHaveAttribute('href', url);
  });

  it('Snapshot test should pass', () => {
    const snap = renderer
      .create(
        <ApplicationThemeProvider theme={defaultTheme}>
          <PolicyText text={text} linkText={linkText} linkUrl={url} />
        </ApplicationThemeProvider>
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
});
