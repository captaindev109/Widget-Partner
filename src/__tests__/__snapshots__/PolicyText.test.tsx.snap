// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases of the PolicyText component Snapshot test should pass 1`] = `
<p
  className="Text__StyledText-sc-1jfsuh1-0 ffZyoX"
  color="gray500"
  data-testid="policy-text"
>
  text
  <a
    className="LinkText-sc-1dzt7t6-0 jA-DDB"
    data-testid="policy-link-text"
    fontSize="12px"
    href="https://www.hedvig.com/se/personuppgifter"
    rel="noopener noreferrer"
    target="_blank"
  >
     linkText
  </a>
  .
</p>
`;
